packages/hadron-build/commands/upload.js

```diff
@@ -1,63 +1,285 @@
   });
 
-  const repo = new GithubRepo({
-    owner: 'mongodb-js',
-    repo: 'compass'
-  }, octokit);
+  const repo = new GithubRepo(
+    {
+      owner: 'mongodb-js',
+      repo: 'compass'
+    },
+    octokit
+  );
 
-  const releaseTag = `v${target.version}`;
+  const releaseTag = `v${version}`;
 
   const release = {
-    name: target.version,
+    name: version,
     tag: releaseTag,
-    notes: `Release v${target.version}`
+    notes: `Release v${version}`
   };
 
-  // NOTE: this will correctly fail if not in draft, so it won't override
-  // already published releases.
-  await repo.updateDraftRelease(release);
+  cli.info('Updating draft release…');
+  JSON.stringify(release, null, 2)
+    .split('\n')
+    .forEach((line) => {
+      cli.info(line);
+    });
+
+  if (!dryRun) {
+    // NOTE: This will correctly fail if not in draft, so it won't override
+    // already published releases.
+    await repo.updateDraftRelease(release);
+  }
+
+  const assetsToUpload = assets
+    .map((item) => {
+      return item.assets;
+    })
+    .flat();
+
+  cli.info('Uploading assets to GitHub release…');
+
+  await checkAssetsExist(
+    assetsToUpload.map((asset) => {
+      return asset.path;
+    })
+  );
 
-  const uploads = target.assets.map(
-    async function(asset) {
-      cli.info(`${asset.name}: upload to Github release ${releaseTag} started (path: ${asset.path}).`);
+  const uploads = assetsToUpload.map(async(asset) => {
+    cli.info(
+      `${
+        asset.name
+      }: upload to Github release ${releaseTag} started (path: ${path.relative(
+        root,
+        asset.path
+      )}).`
+    );
+    if (!dryRun) {
       await repo.uploadReleaseAsset(releaseTag, {
         name: asset.name,
         path: asset.path
       });
-
-      cli.info(`${asset.name}: upload to Github release ${releaseTag} completed.`);
     }
+    cli.info(
+      `${asset.name}: upload to Github release ${releaseTag} completed.`
+    );
+  });
+
+  await Promise.all(uploads);
+}
+
+async function uploadAssetsToDownloadCenter(assets, channel, dryRun) {
+  const assetsToUpload = assets
+    .map((item) => {
+      return item.assets;
+    })
+    .flat()
+    // eslint-disable-next-line no-shadow
+    .filter(({ downloadCenter }) => {
+      return downloadCenter;
+    });
+
+  cli.info('Uploading assets to download center…');
+
+  await checkAssetsExist(
+    assetsToUpload.map((asset) => {
+      return asset.path;
+    })
   );
 
+  const uploads = assetsToUpload.map(async(asset) => {
+    cli.info(
+      `${asset.name}: upload to download center started (path: ${path.relative(
+        root,
+        asset.path
+      )}).`
+    );
+    if (!dryRun) {
+      await uploadAsset(channel, asset);
+    }
+    cli.info(`${asset.name}: upload to download center completed.`);
+  });
+
   await Promise.all(uploads);
 }
 
+async function updateManifest(assets, version, channel, dryRun) {
+  const currentManifest = await downloadManifest();
```

Comment:
Not necessary for this PR, but I think we can improve things further.

It would be amazing to have a "sync" function with what's published on Github rather than "upload(version)".

That's a function that you would be able to call also to "unpublish" a release and wouldn't depend on what was uploaded before.

Reply:
Btw here we may want to check that the version we update is more recent than the previous one, i guess is not super common, but since it doesn't take the latest, if i unpublish and re-publish an old version in github that would replace the latest one right?

Reply:
Just occurred to me that we could do something different:

- as part of EVG upload, we upload a `manifest.json` to Github and the hadron-build job ends here
- on the github workflow each time a release is published or unpublished we get the manifest from the latest GA and latest beta on github, merge them and upload it to S3

Reply:
Ah, I actually like this idea a lot and it wouldn't be that hard to refactor what I have here already to this, let's do it that way!

Reply:
Updated to upload manifest as part of gh release and download latest stable / beta to build updated manifest on manifest update command